webapi/src/main/scala/org/knora/webapi/responders/v2/SearchResponderV2.scala

```diff
@@ -570,324 +570,373 @@ class SearchResponderV2 extends ResponderWithStandoffV2 {
 
                 case filterCompare: CompareExpression =>
 
-                    // left argument of a CompareExpression is expected to be a QueryVariable
-                    val queryVar: QueryVariable = filterCompare.leftArg match {
+                    // left argument of a CompareExpression is expected to be a QueryVariable or a 'lang' function call
+                    filterCompare.leftArg match {
```

Comment:
The method `transformFilterExpression` contains many long, deeply nested `match`-`case` expressions, which makes it very difficult to read. Could you refactor this method so that each `match`-`case` would fit on the screen? You could put each of these into its own method:

* `case filterCompare: CompareExpression =>`
  * `case queryVar: QueryVariable =>`
    * `case propInfo: PropertyTypeInfo =>`
      * `case iriRef: IriRef =>`
    * `case nonPropInfo: NonPropertyTypeInfo =>`
      * `case OntologyConstants.Xsd.Integer =>`
      * `case OntologyConstants.Xsd.Decimal =>`
      * `case OntologyConstants.Xsd.Boolean =>`
      * `case OntologyConstants.Xsd.String =>`
      * `case OntologyConstants.Xsd.Uri =>`
      * `case OntologyConstants.KnoraApiV2Simple.Date =>`
        * `case CompareExpressionOperator.EQUALS =>`
        * `case CompareExpressionOperator.NOT_EQUALS =>`
        * `case CompareExpressionOperator.LESS_THAN =>`
        * `case CompareExpressionOperator.LESS_THAN_OR_EQUAL_TO =>`
        * `case CompareExpressionOperator.GREATER_THAN =>`
        * `case CompareExpressionOperator.GREATER_THAN_OR_EQUAL_TO =>`
  * `case lang: LangFunction =>`
* `case filterOr: OrExpression =>`
* `case filterAnd: AndExpression =>`
* `case regexFunction: RegexFunction =>`
* `case functionCall: FunctionCallExpression =>`

Reply:
@benjamingeer 

>  Could you refactor this method so that each match-case would fit on the screen?

Couldn't this be solved by buying a bigger screen for you? :-)

Just joking, I will do it ;-)

Reply:
@benjamingeer I did the refactoring. Could please look at it again?